tests/system/test_integration.py

```diff
@@ -102,6 +103,40 @@ def test_load_docs_with_template_and_add_error(self):
 
         self.check_backend_error_sourcemap(self.index_error, count=4)
 
+    def test_jaeger_http(self):
+        """
+        This test sends a Jaeger span in Thrift encoding over HTTP, and verifies that it is indexed.
+        """
+        jaeger_span_thrift = self.get_testdata_path('jaeger', 'span.thrift')
+        self.load_docs_with_template(jaeger_span_thrift, self.jaeger_http_url, 'transaction', 1,
+                                     extra_headers={"content-type": "application/vnd.apache.thrift.binary"})
+        self.assert_no_logged_warnings()
+
+        # compare existing ES documents for errors with new ones
+        rs = self.es.search(index=self.index_transaction)
+        assert rs['hits']['total']['value'] == 1, "found {} documents".format(rs['count'])
+        self.approve_docs('jaeger_thrift_span', rs['hits']['hits'], 'transaction')
+
+    def test_jaeger_grpc(self):
+        """
+        This test sends a Jaeger batch over gRPC, and verifies that the spans are indexed.
+        """
+        jaeger_request_data = self.get_testdata_path('..', 'beater', 'jaeger', 'testdata', 'batch_0.json')
+
+        client = os.path.join(os.path.dirname(__file__), 'jaegergrpc')
+        subprocess.check_call(['go', 'run', client,
+            '-addr', self.jaeger_grpc_host,
+            '-insecure',
+            jaeger_request_data,
+        ])
+        self.assert_no_logged_warnings()
+        self.wait_for_events('transaction', 1)
+
+        # compare existing ES documents for errors with new ones
+        rs = self.es.search(index=self.index_transaction)
+        assert rs['hits']['total']['value'] == 1, "found {} documents".format(rs['count'])
+        self.approve_docs('jaeger_batch_0', rs['hits']['hits'], 'transaction')
```

Comment:
Did you leave out the error documents resulting from the `batch_0.json` input on purpose?

Reply:
Kinda, just to get the PR up for initial review. I've added them in now.